api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -598,7 +598,10 @@ private void checkUserCreationNonce(DbUser user, String nonce) {
       UpdateContactEmailRequest updateContactEmailRequest) {
     String username = updateContactEmailRequest.getUsername().toLowerCase();
     com.google.api.services.directory.model.User googleUser = directoryService.getUser(username);
-    DbUser user = userDao.findUserByUsername(username);
+    DbUser user =
+        userService
+            .getByUsername(username)
+            .orElseThrow(() -> new NotFoundException("User '" + username + "' not found"));
```

Comment:
super-nit, but we could centralize the not found error message function and exception function somewhere, like on the suer service, so it'd be `.orElseThrow(() -> userNotFoundEx(username))` and all the messages would match.

Maybe a job for an `Exceptions` utility class.